### PR TITLE
WIP: Update GitHub Actions for PR, Push and Release Cycle

### PR DESCRIPTION
**Pull request by enkeys**

### CODE REVIEW

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
-name: deploy-docs
+---
```
Comment (jiridanek) [inline .github/workflows/docs.yaml:1]: btw, what is this `---` for? Is it useful only if I want to cat multiple yamls to single file?

File: .github/workflows/master.pull_request.workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        run: tox
+      - name: "Build package"
+        run: |
+          sed -i "s/VERSION =.*/VERSION = '${GITHUB_REF##*/}-$GITHUB_SHA'/g" ./src/yacfg/meta.py
+          python setup.py sdist bdist_wheel
+      - name: "Upload artifacts"
+        uses: actions/upload-artifact@v2
+        with:
+          name: package-dist
+          path: dist/
```
Comment (jiridanek) [inline .github/workflows/master.pull_request.workflow.yaml:42]: newline?

File: .github/workflows/master.push.workflow.yaml [hunk 3]
```diff
@@ -31,6 +30,22 @@ jobs:
...
+        with:
+          name: package-dist
+          path: dist/
+      - name: "Publish package to TestPyPI"
+        uses: pypa/gh-action-pypi-publish@v1.3.1
+        with:
+          skip_existing: true
+          user: __token__
+          password: ${{ secrets.TESTPYPI_TOKEN }}
+          repository_url: https://test.pypi.org/legacy/
```
Comment (jiridanek) [inline .github/workflows/master.push.workflow.yaml:51]: newline

File: .github/workflows/stable.release.workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          skip_existing: true
+          user: __token__
+          password: ${{ secrets.PYPI_TOKEN }}
+      - name: "Upload Release Asset"
+        id: upload-release-asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
```
Comment (jiridanek) [inline .github/workflows/stable.release.workflow.yaml:55]: looong line

File: .github/workflows/docs.yaml [hunk 5]
```diff
@@ -1,4 +1,5 @@
...
+name: "Deploy docs"
 
 # Only run this when the master branch changes
```
Comment (jiridanek) [inline .github/workflows/docs.yaml:4]: any way to allow doc preview for prs, or maybe do the doc build without the push, to check build is not broken? I am thinking about this UI element on github

image

**Review by jiridanek — APPROVED** [review]

File: .github/workflows/docs.yaml (hunk 1 above)
Comment (enkeys) [reply .github/workflows/docs.yaml:1]: We are using yamllint and in default. So '---' it's required for document-start. But anyway it's a good habit for YAML from my point of view. 

More info:
* https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.document_star

File: .github/workflows/docs.yaml (hunk 5 above)
Comment (enkeys) [reply .github/workflows/docs.yaml:4]: Yes, I can add build-docs without pushing to gh-pages I hope that is a better approach than pushing all changes from the PR's. It's ok?

**Review by zkraus — APPROVED** [review]

File: .github/workflows/docs.yaml (hunk 5 above)
Comment (jiridanek) [reply .github/workflows/docs.yaml:4]: sounds good

Comment (enkeys) [reply .github/workflows/docs.yaml:4]: Okay, so we have now built docs with tox. For every PR and master push is uploaded docs/_build/html as docs_build zip artifact.

**Review by zkraus — APPROVED** [review]

**Review by zkraus — APPROVED** [review]

**Comment by enkeys** [comment]
Isort was dropped, will do in next PR.

**Pull request merged**